openslide/__init__.py

```diff
@@ -149,7 +149,9 @@ class OpenSlide(AbstractSlide):
     def __init__(self, filename):
         """Open a whole-slide image."""
         AbstractSlide.__init__(self)
-        self._filename = filename
+        # Filename might be a pathlib.Path instance, but file reader expects
+        # a string.
+        self._filename = str(filename)
         self._osr = lowlevel.open(filename)
```

Comment:
The problem with the pathlib.Path instance is actually where filename is passed to `lowlevel.open()` and also in the following detect_format classmethod where a filename is passed to `lowlevel.detect_vendor()`.

Updating the tests to catch functions being called with Path instances is useful to catch regressions here. I changed `tests/__init__.py:file_path` to `return pathlib.Path(__file__).parent / name` and it found a lot of typeerrors while running the tests, but those all originated from those two `lowlevel.open` and `lowlevel.detect_vendor` calls.

Reply:
Thanks for the comment. What if we also converted the input to `lowlevel.detect_format()` to a `str`? I agree that there should be a test using `pathlib.Path` instances.

Reply:
I think I can push my updates directly to your branch if 'allow edits by maintainers' is enabled on your PR.